app/src/webtrigger/handle-reset-jenkins-request.test.ts

```diff
@@ -14,8 +14,12 @@ jest.mock('../jira-client/delete-deployments');
 const CLOUD_ID = '97eaf652-4b6e-46cf-80c2-d99327a63bc1';
 
 describe('Reset Jenkins Server Suite', () => {
-	when(disconnectJenkinsServer).mockImplementation(() => Promise.resolve(true));
-	when(getAllJenkinsServers).mockImplementation(() => Promise.resolve([mockSingleJenkinsPipeline]));
+
+	beforeEach(() => {
+		jest.clearAllMocks();
```

Comment:
This should be done in an `afterEach`, not `beforeEach`.  You can also make this global if you use a setup file for jest so that _all tests_ clear their mocks after they're done.